### PR TITLE
nixos/calibre-server: Add address configuration

### PR DESCRIPTION
**Pull request by endocrimes**
This commit introduces two new fields into the configuration of calibre-server, listenAddress and port. These allow the operator to configure the addresses that calibre should listen on, and the port it should bind to respectively.

The default values here are copied from calibre itself, to avoid backwards compatibility issues or more complicated service configuration.

###### Motivation for this change

calibre defaults to listening on port 8080 on all interfaces, which isn't ideal in almost any configuration outside of developing calibre itself. Exposing this configuration lets people run calibre more reasonably.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @domenkozar @pSub @AndersonTorres

### CODE REVIEW

File: nixos/modules/services/misc/calibre-server.nix [hunk 1]
```diff
@@ -13,20 +13,34 @@ in
...
+        '';
+        type = types.path;
       };
 
-    };
+      listenAddress = mkOption {
+        default = "::";
+        example = "127.0.0.1";
+        description = ''
+          The interface on which to listen for connections. The value "::" will
```
Comment (contrun) [inline nixos/modules/services/misc/calibre-server.nix]: Should we change it to `The address on which to listen for connections`

Comment (endocrimes) [reply nixos/modules/services/misc/calibre-server.nix]: Could do - I was keeping it inline with the upstream documentation for now, but don't feel strongly either way.

**Review by spacekookie — APPROVED** [review]

File: nixos/modules/services/misc/calibre-server.nix (hunk 1 above)
Comment (contrun) [reply nixos/modules/services/misc/calibre-server.nix]: OK. Thanks for the explanation.

**Review by contrun — APPROVED** [review]

**Comment by endocrimes** [comment]
Hey @domenkozar/@pSub - Was hoping that one of you may be able to take a look at this soon? Thanks!

**Comment by AndersonTorres** [comment]
@endocrimes PING

**Comment by Petemir** [comment]
@endocrimes Any chance of solving the conflict? Thanks!

**Comment by pSub** [comment]
@AndersonTorres Why are you closing this issue? I tried to resolve the conflicts yesterday. Somehow ofborg fail and I wanted to investigate this weekend.

**Comment by AndersonTorres** [comment]
Two years and the original poster ignored the pings and replies. I closed it because it was abandoned.

If anyone is interested, certainly it can be reopened.

**Pull request closed without merging**